PROVESFlightControllerReference/test/unit-tests/test_FlashWorker_UpdateSequencer.cpp

```
@@ -0,0 +1,263 @@
+    UpdateSequencer sequencer = prepared();
+    EXPECT_EQ(UpdateSequencer::Status::IMAGE_CRC_MISMATCH,
+              sequencer.onUpdateComplete(UpdateSequencer::WriteOutcome::CRC_MISMATCH));
+}
+
+TEST(UpdateSequencerTest, FlashWriteFailureReportsFlashWriteError) {
+    UpdateSequencer sequencer = prepared();
+    EXPECT_EQ(UpdateSequencer::Status::FLASH_WRITE_ERROR,
+              sequencer.onUpdateComplete(UpdateSequencer::WriteOutcome::FLASH_WRITE_FAILED));
+}
+
+TEST(UpdateSequencerTest, NoFailureOutcomeReportsSuccess) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)